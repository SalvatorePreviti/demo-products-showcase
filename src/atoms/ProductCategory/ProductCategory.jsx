import React from 'react'
import PropTypes from 'prop-types'
import styles from './ProductCategory.scss'

const itemStyle = styles.ProductCategory
const itemActiveStyle = `${itemStyle} ${styles.active}`

function ProductCategory(props) {
  const { title, active, onSelect } = props

  const onClick = event => {
    if (onSelect) {
      onSelect()
    }
    if (event) {
      event.preventDefault()
    }
  }

  return (
    <a href="#" className={active ? itemActiveStyle : itemStyle} onClick={onClick}>
      {title}
    </a>
  )
}

ProductCategory.propTypes = {
  title: PropTypes.string,
  active: PropTypes.bool,
  onSelect: PropTypes.func
}

export default ProductCategory
