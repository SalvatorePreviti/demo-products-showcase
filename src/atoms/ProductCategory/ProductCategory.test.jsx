import React from 'react'
import { shallow } from 'enzyme'

import ProductCategory from './ProductCategory'

describe('ProductCategory', () => {
  describe('without selection', () => {
    let root

    beforeAll(() => {
      root = shallow(<ProductCategory title="A title" />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('renders', () => {
      const found = root.find('a')
      expect(found.props()).toMatchObject({ href: '#', className: 'ProductCategory', children: 'A title' })
    })
  })

  describe('with selection', () => {
    let root

    beforeAll(() => {
      root = shallow(<ProductCategory title="An active item" active={true} />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('renders', () => {
      const found = root.find('a')
      expect(found.props()).toMatchObject({ href: '#', className: 'ProductCategory active', children: 'An active item' })
    })
  })

  describe('when clicking', () => {
    it('invokes onSelectCategory with the category id', () => {
      let clicked = false
      const root = shallow(
        <ProductCategory
          title="title"
          onSelect={() => {
            clicked = true
          }}
        />
      )

      root.simulate('click')

      expect(clicked).toBe(true)
    })
  })
})
