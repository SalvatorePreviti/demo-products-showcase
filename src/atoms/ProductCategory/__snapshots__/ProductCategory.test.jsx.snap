// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategory with selection snapshots 1`] = `
<a
  className="ProductCategory active"
  href="#"
  onClick={[Function]}
>
  An active item
</a>
`;

exports[`ProductCategory without selection snapshots 1`] = `
<a
  className="ProductCategory"
  href="#"
  onClick={[Function]}
>
  A title
</a>
`;
